messaginginapp/src/main/java/io/customer/messaginginapp/ui/core/BaseInlineInAppMessageView.kt

```diff
@@ -0,0 +1,65 @@
+package io.customer.messaginginapp.ui.core
+
+import android.content.Context
+import android.util.AttributeSet
+import android.widget.FrameLayout
+import androidx.annotation.AttrRes
+import androidx.annotation.StyleRes
+import io.customer.base.internal.InternalCustomerIOApi
+import io.customer.messaginginapp.type.InlineMessageActionListener
+import io.customer.messaginginapp.ui.bridge.InAppHostViewDelegateImpl
+import io.customer.messaginginapp.ui.bridge.InAppPlatformDelegate
+import io.customer.messaginginapp.ui.bridge.InlineInAppMessageViewCallback
+import io.customer.messaginginapp.ui.controller.InlineInAppMessageViewController
+
+/**
+ * Abstract base class for inline in-app message views.
+ *
+ * Provides core functionality for displaying inline messages including element ID management
+ * and action listener support. Subclasses must implement [platformDelegate] to provide
+ * platform-specific rendering and interaction handling.
+ */
+abstract class BaseInlineInAppMessageView<P : InAppPlatformDelegate>
+@InternalCustomerIOApi
+@JvmOverloads
+constructor(
+    context: Context,
+    attrs: AttributeSet? = null,
+    @AttrRes defStyleAttr: Int = 0,
+    @StyleRes defStyleRes: Int = 0
+) : FrameLayout(context, attrs, defStyleAttr, defStyleRes), InlineInAppMessageViewCallback {
+    protected abstract val platformDelegate: P
+    internal val controller: InlineInAppMessageViewController by lazy {
+        InlineInAppMessageViewController(
+            viewDelegate = InAppHostViewDelegateImpl(view = this),
+            platformDelegate = platformDelegate
+        )
+    }
+
+    @InternalCustomerIOApi
+    protected fun configureView() {
```

Comment:
we made it manual instead of automatically initializing it in `init` due to lifecycle difference in wrapper?

Reply:
Yes, and to access `this` safely in `init` of final view